app/identifiers/register/trustees/partnership/PartnershipAddressId.scala

```diff
@@ -30,6 +32,19 @@ case class PartnershipAddressId(index: Int) extends TypedIdentifier[Address] {
 object PartnershipAddressId {
   override def toString: String = "partnershipAddress"
 
-  implicit def cya(implicit countryOptions: CountryOptions): CheckYourAnswers[PartnershipAddressId] =
-    AddressCYA[PartnershipAddressId](changeAddress = "messages__visuallyhidden__partnership__address")()
+  implicit def cya(implicit countryOptions: CountryOptions, messages: Messages, ua: UserAnswers): CheckYourAnswers[PartnershipAddressId] =
+
+    new CheckYourAnswers[PartnershipAddressId] {
+      override def row(id: PartnershipAddressId)(changeUrl: String, ua: UserAnswers): Seq[AnswerRow] = {
+        def trusteeName(index: Int) = ua.get(PartnershipDetailsId(index)).fold(messages("messages__theTrustee"))(_.name)
+        def label(index: Int) = messages("messages__addressFor", trusteeName(index))
```

Comment:
label can be a val instead of def as index is already available inside row

Reply:
fixed